pkg/types/builder.go

```diff
@@ -39,28 +39,38 @@ const (
 // NewBuilder returns a new Builder.
 func NewBuilder(pkg *types.Package) *Builder {
 	return &Builder{
-		Package:  pkg,
-		comments: twtypes.Comments{},
+		Package: pkg,
+		Generated: Generated{
+			Comments: twtypes.Comments{},
+		},
 	}
 }
 
+// Generated stores the information of the generated types
+type Generated struct {
+	Types    []*types.Named
+	Comments twtypes.Comments
+
+	ForProviderType *types.Named
+	AtProviderType  *types.Named
+}
+
 // Builder is used to generate Go type equivalence of given Terraform schema.
 type Builder struct {
 	Package *types.Package
-
-	genTypes []*types.Named
```

Comment:
Originally, I'd have preferred to not have them as properties at all since they are very specific to a single call of `Build` but I had to do it so that recursive calls can access it without having to pass them around all the time. What do you think about keeping these private and return a `Generated` instance that is created and filled up in `Build` as the result?